src/app/guards/security/is-logged-in/is-logged-in.guard.ts

```diff
@@ -0,0 +1,17 @@
+import { inject } from "@angular/core";
+import { CanActivateFn, Router } from "@angular/router";
+import { BawSessionService } from "@baw-api/baw-session.service";
+import { loginRoute } from "@components/security/security.routes";
+
+export const isLoggedInGuard: CanActivateFn = () => {
+  const sessionService = inject(BawSessionService);
+  const router = inject(Router);
+
+  if (sessionService.isLoggedIn) {
+    return true;
+  }
+
+  // by returning a url tree from this guard, the guard will navigate to the
+  // login page if the user is not logged in.
+  return router.createUrlTree([loginRoute.toRouterLink()]);
```

Comment:
can we set a redirect back to the current page after log in?

Reply:
Associated commit [`0a80b19` (#2305)](https://github.com/QutEcoacoustics/workbench-client/pull/2305/commits/0a80b1971f175f062f095afbdebb7bff076128dc)

I had to fix the login page redirect preferring history.back over the `?redirect` url parameter.